Userland/Libraries/LibWeb/PerformanceTimeline/PerformanceObserverEntryList.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2023, Luke Wilde <lukew@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibWeb/Bindings/PlatformObject.h>
+
+namespace Web::PerformanceTimeline {
+
+// https://w3c.github.io/performance-timeline/#performanceobserverentrylist-interface
+class PerformanceObserverEntryList final : public Bindings::PlatformObject {
+    WEB_PLATFORM_OBJECT(PerformanceObserverEntryList, Bindings::PlatformObject);
+
+public:
+    virtual ~PerformanceObserverEntryList() override;
+
+    WebIDL::ExceptionOr<Vector<JS::Handle<PerformanceTimeline::PerformanceEntry>>> get_entries() const;
+    WebIDL::ExceptionOr<Vector<JS::Handle<PerformanceTimeline::PerformanceEntry>>> get_entries_by_type(String const& type) const;
+    WebIDL::ExceptionOr<Vector<JS::Handle<PerformanceTimeline::PerformanceEntry>>> get_entries_by_name(String const& name, Optional<String> type) const;
+
+private:
+    PerformanceObserverEntryList(JS::Realm&, Vector<JS::Handle<PerformanceTimeline::PerformanceEntry>>&&);
+
+    virtual void initialize(JS::Realm&) override;
+
+    // https://w3c.github.io/performance-timeline/#dfn-entry-list
+    // Returns a PerformanceEntryList object returned by filter buffer by name and type algorithm with this's entry list,
+    // name and type set to null.
+    Vector<JS::Handle<PerformanceTimeline::PerformanceEntry>> m_entry_list;
```

Comment:
can we make it a vector of `JS::GCPtr` and visit it in `visit_edges` instead of using `JS::Handle`? I had been working recently on fixing GC leaks and found that `JS::Handle` that introduces new heap roots makes it quite harder to understand what is going on.